src/common/com/intellij/plugins/haxe/ide/index/HaxeComponentFileNameIndex.java

```diff
@@ -99,17 +101,26 @@ public static boolean getFileNames(@NotNull String qName,
     @NotNull
     public Map<String, Void> map(final FileContent inputData) {
       final PsiFile psiFile = inputData.getPsiFile();
+      final String moduleName = FileUtil.getNameWithoutExtension(inputData.getFileName());
       final List<HaxeClass> classes = HaxeResolveUtil.findComponentDeclarations(psiFile);
       if (classes.isEmpty()) {
         return Collections.emptyMap();
```

Comment:
If there is no class which matches filename - then it would be added to result, like in haxe.macro.Tools there is no class which matches filename